### PR TITLE
expression: reorganize error messages for non-existent functions

### PR DESCRIPTION
**Pull request by zz-jason**
to #4373

### CODE REVIEW

**Comment by zz-jason** [comment]
"unit-test" failed due to "data race" in go leveldb, @tiancaiamao is working on it in #4543

**Comment by XuHuaiyu** [comment]
LGTM

File: expression/errors.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		mysql.ErrZlibZData:                  mysql.ErrZlibZData,
+		mysql.ErrWrongArguments:             mysql.ErrWrongArguments,
+	}
+	terror.ErrClassToMySQLCodes[terror.ClassExpression] = expressionMySQLErrCodes
+}
+
+// handleInvalidTimeError reports error or warning depend on the context.
+func handleInvalidTimeError(ctx context.Context, err error) error {
+	if err == nil || !(terror.ErrorEqual(err, types.ErrInvalidTimeFormat) || types.ErrIncorrectDatetimeValue.Equal(err)) {
+		return err
```
Comment (zimulala) [inline expression/errors.go]: Add the error trace?

Comment (zz-jason) [reply expression/errors.go]: no, this function should never touch the input error, it just decides whether to append the input error to warnings.

**Pull request merged**